src/index.ts

```diff
@@ -24,6 +24,39 @@ type LooseNode = Node & {
   value: any;
 };
 
+const generateHeaders = async () => {
+  await writeFile(
+    ".vercel/output/static/_headers",
+    `\n/_next/static/*\n  Cache-Control: public,max-age=31536000,immutable\n`,
```

Comment:
Two things on this `_headers` generation:

1. `.vercel/output/config.json` should include a route rule which covers this. For example, in a build I just produced, I find the following:

    ```json
    {
      "routes": [
        {
          "src": "/_next/static/(?:[^/]+/pages|pages|chunks|runtime|css|image|media|3xPbVjttysUBv9VjNExZE)/.+",
          "headers": {
            "cache-control": "public,max-age=31536000,immutable"
          },
          "continue": true,
          "important": true
        }
      ]
    }
    ```

    It would be great to pull from here and just automatically apply all the headers for all the rules that this gives us. Perhaps this is best tackled in a separate PR? I won't personally block on this though, so we all want to get the current implementation through on this current PR, it's okay with me, but I'd at least like to create an issue to track the follow-up of implementing this from `.vercel/output/config.json` instead.

1. [SvelteKit also produce a `_headers` file](https://github.com/sveltejs/kit/blob/c45c35e7b40837cd1bea3fc26cfb831478e03c00/packages/adapter-cloudflare/index.js#L133). They add a nice comment wrapping the autogenerated rules. It would be nice to do the same here for any human that comes along and inspects it.

Reply:
> It would be great to pull from here and just automatically apply all the headers for all the rules that this gives us.

Yeah I was a little wary of that since they limit their paths much more aggressively and I didn't want to expand (potentially) unsafe headers to arbitrary paths. Can change that though if we assume it's safe enough.

Reply:
@Sebmaster I think it should be safe enough and that does sound like the best way of implementing this

So I think we can either merge this as is and open an issue/followup PR to apply the improvement or change this before merging

If it's quick enough let's do it here otherwise let's open an issue about it

what do you think? could you have a look and let me know which is? 🙂 🙏